Libs/MRML/Widgets/qMRMLSliceControllerWidget.cxx

```diff
@@ -2306,6 +2368,58 @@ void qMRMLSliceControllerWidget::showReformatWidget(bool show)
     }
 }
 
+//---------------------------------------------------------------------------
+void qMRMLSliceControllerWidget::showReconstructionWidget(bool show)
+{
+  Q_D(qMRMLSliceControllerWidget);
+
+  vtkSmartPointer<vtkCollection> nodes = d->saveNodesForUndo("vtkMRMLSliceNode");
+  if (!nodes.GetPointer())
+    {
+    return;
+    }
+  vtkMRMLSliceNode* node = nullptr;
+  vtkMRMLSliceLogic* sliceLogic = nullptr;
+  for (int i = 0; i < nodes->GetNumberOfItems(); i++)
+  {
+    node = static_cast<vtkMRMLSliceNode*>(nodes->GetItemAsObject(i));
+    sliceLogic = static_cast<vtkMRMLSliceLogic*>(d->SliceLogics->GetItemAsObject(i));
+    if (node == this->mrmlSliceNode() || this->isLinked())
+      {
+        node->SetSlabReconstructionVisibility(show);
+        vtkImageReslice* resliceBG = sliceLogic->GetBackgroundLayer()->GetReslice();
+        vtkImageReslice* resliceFG = sliceLogic->GetForegroundLayer()->GetReslice();
+        if (show)
+        {
+          int slabType = node->GetSlabReconstructionType();
+          resliceBG->SetSlabMode(slabType);
+          resliceFG->SetSlabMode(slabType);
+
+          vtkMRMLVolumeNode* volumeBG = sliceLogic->GetLayerVolumeNode(vtkMRMLSliceLogic::LayerBackground);
+          if (volumeBG)
+          {
+            double maxVolumeSpacingBG = volumeBG->GetMaxSpacing();
```

Comment:
I don't `MaxSpacing` has any relevance here. Reslicing uses the output spacing as a basis for slice thickness, not the maximum spacing of the input volume.

Also, most of the time we would want to take multiple slices, even if the slab thickness is less than a the slice thickness, because the output voxel values are computed with interpolation (so the image intensity value is not constant within half slice distance, but it is gradually changing).

Also note that everything has to work well for oblique slices. Spacing for oblique slices is computed in vtkMRMLSliceLogic::GetVolumeSliceSpacing - it matches slice spacing along an axis if the slice normal is aligned with an axis and it is smoothly interpolated if the slice is oblique.

Reply:
`MasSpacing` originally came from [QReads](https://github.com/KitwareMedical/SlicerQReads/blob/d40ce875aa49f396fc8fe2b578e477d6cd02894f/Modules/Scripted/QReads/QReads.py#L609-L613) implementation of converting thickness in mm to number of slices.

If `MaxSpacing` is removed, would the `SlabThicknessinMM` be the input or is there a better computation?

Reply:
> Reslicing uses the output spacing as a basis for slice thickness

This makes sense. To provide some context, the `QReads` implementation was done in the context of a prototype and some shortcut were taken.